File: 3191. Minimum Operations to Make Binary Array Elements Equal to One I/main.go
package main

func minOperations(nums []int) int {
	res := 0
	for i := range len(nums) - 2 {
		if nums[i] == 0 {
			for j := i; j < i+3; j++ {
				if nums[j] == 0 {
					nums[j] = 1
				} else {
					nums[j] = 0
				}
			}
			res++
		}
	}
	if nums[len(nums)-1] == 0 || nums[len(nums)-2] == 0 {
		return -1
	}
	return res
}

File: 3191. Minimum Operations to Make Binary Array Elements Equal to One I/main.py
from typing import List


class Solution:
    def minOperations(self, nums: List[int]) -> int:
        res = 0
        for i in range(len(nums) - 2):
            if nums[i] == 0:
                for j in range(i, i + 3):
                    nums[j] = 0 if nums[j] else 1
                res += 1
        if not nums[-1] or not nums[-2]:
            return -1
        return res